### ISSUE TITLE
can you help me

### ISSUE THREAD

**Issue by vulee123** [thejoabo/virtualfisher-bot#37]
[*] Starting...
[*] Session stablished !

            DEBUG -> Exception - Response

            TRACEBACK -> [FrameInfo(frame=<frame at 0x0000017D84340370, file 'C:\\Users\\Anh Vu\\OneDrive\\Desktop\\virtualfisher-bot-main\\autofishbot.py', line 467, code message_receiver>, filename='C:\\Users\\Anh Vu\\OneDrive\\Desktop\\virtualfisher-bot-main\\autofishbot.py', lineno=456, function='message_receiver', code_context=['            response = session.receive_event()\n'], index=0), FrameInfo(frame=<frame at 0x0000017D848A9D80, file 'C:\\Users\\Anh Vu\\OneDrive\\Desktop\\virtualfisher-bot-main\\app\\DiscordWrapper.py', line 272, code receive_event>, filename='C:\\Users\\Anh Vu\\OneDrive\\Desktop\\virtualfisher-bot-main\\app\\DiscordWrapper.py', lineno=272, function='receive_event', code_context=['        _response = self.ws.recv()\n'], index=0), FrameInfo(frame=<frame at 0x0000017D848B03E0, file 'C:\\Users\\Anh Vu\\AppData\\Local\\Programs\\Python\\Python310\\lib\\site-packages\\websocket\\_core.py', line 359, code recv>, filename='C:\\Users\\Anh Vu\\AppData\\Local\\Programs\\Python\\Python310\\lib\\site-packages\\websocket\\_core.py', lineno=359, function='recv', code_context=['            opcode, data = self.recv_data()\n'], index=0), FrameInfo(frame=<frame at 0x0000017D84789EA0, file 'C:\\Users\\Anh Vu\\AppData\\Local\\Programs\\Python\\Python310\\lib\\site-packages\\websocket\\_core.py', line 382, code recv_data>, filename='C:\\Users\\Anh Vu\\AppData\\Local\\Programs\\Python\\Python310\\lib\\site-packages\\websocket\\_core.py', lineno=382, function='recv_data', code_context=['        opcode, frame = self.recv_data_frame(control_frame)\n'], index=0), FrameInfo(frame=<frame at 0x0000017D84765E00, file 'C:\\Users\\Anh Vu\\AppData\\Local\\Programs\\Python\\Python310\\lib\\site-packages\\websocket\\_core.py', line 403, code recv_data_frame>, filename='C:\\Users\\Anh Vu\\AppData\\Local\\Programs\\Python\\Python310\\lib\\site-packages\\websocket\\_core.py', lineno=403, function='recv_data_frame', code_context=['            frame = self.recv_frame()\n'], index=0), FrameInfo(frame=<frame at 0x0000017D8455B0B0, file 'C:\\Users\\Anh Vu\\AppData\\Local\\Programs\\Python\\Python310\\lib\\site-packages\\websocket\\_core.py', line 442, code recv_frame>, filename='C:\\Users\\Anh Vu\\AppData\\Local\\Programs\\Python\\Python310\\lib\\site-packages\\websocket\\_core.py', lineno=442, function='recv_frame', code_context=['        return self.frame_buffer.recv_frame()\n'], index=0), FrameInfo(frame=<frame at 0x0000017D823A43F0, file 'C:\\Users\\Anh Vu\\AppData\\Local\\Programs\\Python\\Python310\\lib\\site-packages\\websocket\\_abnf.py', line 338, code recv_frame>, filename='C:\\Users\\Anh Vu\\AppData\\Local\\Programs\\Python\\Python310\\lib\\site-packages\\websocket\\_abnf.py', lineno=338, function='recv_frame', code_context=['                self.recv_header()\n'], index=0), FrameInfo(frame=<frame at 0x0000017D8433A950, file 'C:\\Users\\Anh Vu\\AppData\\Local\\Programs\\Python\\Python310\\lib\\site-packages\\websocket\\_abnf.py', line 294, code recv_header>, filename='C:\\Users\\Anh Vu\\AppData\\Local\\Programs\\Python\\Python310\\lib\\site-packages\\websocket\\_abnf.py', lineno=294, function='recv_header', code_context=['        header = self.recv_strict(2)\n'], index=0), FrameInfo(frame=<frame at 0x0000017D848B05B0, file 'C:\\Users\\Anh Vu\\AppData\\Local\\Programs\\Python\\Python310\\lib\\site-packages\\websocket\\_abnf.py', line 373, code recv_strict>, filename='C:\\Users\\Anh Vu\\AppData\\Local\\Programs\\Python\\Python310\\lib\\site-packages\\websocket\\_abnf.py', lineno=373, function='recv_strict', code_context=['            bytes_ = self.recv(min(16384, shortage))\n'], index=0), FrameInfo(frame=<frame at 0x0000017D848B0780, file 'C:\\Users\\Anh Vu\\AppData\\Local\\Programs\\Python\\Python310\\lib\\site-packages\\websocket\\_core.py', line 532, code _recv>, filename='C:\\Users\\Anh Vu\\AppData\\Local\\Programs\\Python\\Python310\\lib\\site-packages\\websocket\\_core.py', lineno=526, function='_recv', code_context=['            return recv(self.sock, bufsize)\n'], index=0), FrameInfo(frame=<frame at 0x0000017D8433B3A0, file 'C:\\Users\\Anh Vu\\AppData\\Local\\Programs\\Python\\Python310\\lib\\site-packages\\websocket\\_socket.py', line 122, code recv>, filename='C:\\Users\\Anh Vu\\AppData\\Local\\Programs\\Python\\Python310\\lib\\site-packages\\websocket\\_socket.py', lineno=122, function='recv', code_context=['        raise WebSocketConnectionClosedException(\n'], index=0)]

            EVENT -> Connection to remote host was lost.

            STACK -> [FrameInfo(frame=<frame at 0x0000017D848BC040, file 'C:\\Users\\Anh Vu\\OneDrive\\Desktop\\virtualfisher-bot-main\\app\\Util.py', line 18, code debug>, filename='C:\\Users\\Anh Vu\\OneDrive\\Desktop\\virtualfisher-bot-main\\app\\Util.py', lineno=22, function='debug', code_context=['            STACK -> {stack()}\\n\n'], index=0), FrameInfo(frame=<frame at 0x0000017D84340370, file 'C:\\Users\\Anh Vu\\OneDrive\\Desktop\\virtualfisher-bot-main\\autofishbot.py', line 467, code message_receiver>, filename='C:\\Users\\Anh Vu\\OneDrive\\Desktop\\virtualfisher-bot-main\\autofishbot.py', lineno=467, function='message_receiver', code_context=["            debugger.debug(e, 'Exception - Response')\n"], index=0), FrameInfo(frame=<frame at 0x0000017D84766A40, file 'C:\\Users\\Anh Vu\\AppData\\Local\\Programs\\Python\\Python310\\lib\\threading.py', line 946, code run>, filename='C:\\Users\\Anh Vu\\AppData\\Local\\Programs\\Python\\Python310\\lib\\threading.py', lineno=946, function='run', code_context=['                self._target(*self._args, **self._kwargs)\n'], index=0), FrameInfo(frame=<frame at 0x0000017D8473DF40, file 'C:\\Users\\Anh Vu\\AppData\\Local\\Programs\\Python\\Python310\\lib\\threading.py', line 1009, code _bootstrap_inner>, filename='C:\\Users\\Anh Vu\\AppData\\Local\\Programs\\Python\\Python310\\lib\\threading.py', lineno=1009, function='_bootstrap_inner', code_context=['                self.run()\n'], index=0), FrameInfo(frame=<frame at 0x0000017D8478ADD0, file 'C:\\Users\\Anh Vu\\AppData\\Local\\Programs\\Python\\Python310\\lib\\threading.py', line 966, code _bootstrap>, filename='C:\\Users\\Anh Vu\\AppData\\Local\\Programs\\Python\\Python310\\lib\\threading.py', lineno=966, function='_bootstrap', code_context=['            self._bootstrap_inner()\n'], index=0)]

0 values:
sigma: (0, 0.12289)

[!] User exited.

**Comment by thejoabo** [comment]
Hello @vulee123 , this seems to be a common bug nowadays related to the websocket-client package it just returns an exception with "\n" (which doesn't make sense), if your config (channel id and token) is correctly set up try to change your IP with a VPN or proxy, it appears to be like the websocket is refusing to connect. Since there is no value on cooldown analysis I assume it doesn't even start properly and therefore a resuming attempt wouldn't be ideal.

In summary:
- I'm not handling properly the exception on the discord's wrapper class
- Your connection is being refused

**The only thing I could suggest for now is to try run it over a VPN/proxy**.

(when I have time I'll be rewriting this whole wrapper class)
